airflow/executors/executor_loader.py

```diff
@@ -184,6 +184,22 @@ def get_default_executor(cls) -> BaseExecutor:
 
         return default_executor
 
+    @classmethod
+    def set_default_executor(cls, executor: BaseExecutor) -> None:
```

Comment:
I dont know if it makes sense to unit test this method, but at least I think it'd be worth to cover the method `Dag.run` you modify in this PR as well (when `local` is True)

Reply:
You actually did! My bad :)